packages/openapi-v3/src/__tests__/unit/consolidate-schema.unit.ts

```diff
@@ -0,0 +1,195 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/openapi-v3
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {ConsolidateSchemaObjects, OpenAPIObject} from '../..';
+
+describe('ConsolidateSchemaObjects', () => {
+  it('moves schema with title to component.schemas, replace with reference', () => {
+    const inputSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          title: 'loopback.example',
+          properties: {
+            test: {
+              type: 'string',
+            },
+          },
+        },
+      },
+    };
+    const expectedSpec: OpenAPIObject = {
+      openapi: '',
+      info: {
+        title: '',
+        version: '',
+      },
+      paths: {
+        schema: {
+          $ref: '#/components/schemas/LoopbackExample',
+        },
+      },
+      components: {
+        schemas: {
+          LoopbackExample: {
```

Comment:
What is the reasoning for converting schema titles into different key values (`loopback.example` to `LoopbackExample`)? So far, we are always using the title as the key.

Reply:
I was thinking along the lines of the Openapi spec being consumed by a code generator.  Model names generated by picking up the title and PascalCase was prettier.  It's not essential, can drop.

Reply:
Dropped as unable to justify here.